packages/slo-stat-bot/src/slo-appliesTo.ts

```diff
@@ -0,0 +1,210 @@
+  excludedGitHubLabels.forEach((label: string) => label.toLowerCase());
+  const isElementExist = excludedGitHubLabels.some((label: string) =>
+    issueLabels.includes(label)
+  );
+  return !isElementExist;
+};
+
+/**
+ * Function checks if the rule (priority or type) exists in issue labels
+ * @param issueLabels of the issue
+ * @param rule is either priority or type (ex: bug, enhancement) of issue
+ * @param title of the rule
+ * @returns true if rule applies to issue else false
+ */
+export const isValidRule = async function isValidRule(
+  issueLabels: string[],
+  rule: string | undefined,
+  title: string
+) {
+  if (!rule) {
+    return true;
+  }
+
+  rule = rule.toLowerCase();
+
+  if (issueLabels.includes(rule)) {
+    return true;
+  }
+  if (issueLabels.includes(title + rule)) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Function converts the variable to an array if it is a string
+ * @param variable can either be array or string
+ * @returns an array
+ */
+export const convertToArray = async function convertToArray(
+  variable: string[] | string
+): Promise<string[]> {
+  if (typeof variable === 'string') {
+    return [variable];
+  }
+  return variable;
+};
+
+/**
+ * Function determines if slo applies to the issue
+ * @param type specifies if event is issue or pr
+ * @param slo rules
+ * @param issueLabels of issue or pr
+ * @param number of issue or pr
+ * @returns true if slo applies to issue else false
+ */
+export const doesSloApply = async function doesSloApply(
+  type: string,
+  slo: SLORules,
+  issueLabels: string[] | null,
+  number: number
+): Promise<boolean> {
+  const sloString = JSON.stringify(slo, null, 4);
+
+  if (Object.keys(slo.appliesTo).length === 0) {
+    return true;
+  }
+
+  if (issueLabels === null) {
+    console.info(
+      `Skipping issue ${number} for rule ${sloString} \n as it does not apply`
+    );
+    return false;
+  }
+
+  if (issueLabels.length === 0) {
+    console.info(
+      `Skipping issue ${number} for rule ${sloString} \n as it does not apply`
+    );
+    return false;
+  }
+
+  const appliesToIssues = slo.appliesTo.issues;
+  const appliesToPrs = slo.appliesTo.prs;
+  const appliesToType = await isValidType(appliesToIssues, appliesToPrs, type);
+  if (!appliesToType) {
+    console.info(
+      `Skipping issue ${number} for rule ${sloString} \n as it does not apply to type`
+    );
+    return false;
+  }
+
+  const githubLabels = slo.appliesTo.gitHubLabels;
+  const hasGithubLabels = await isValidGithubLabels(issueLabels, githubLabels);
+  if (!hasGithubLabels) {
+    console.info(`
+    Skipping issue ${number} for rule ${sloString} \n as it does not apply to gitHubLabels`);
+    return false;
+  }
+
+  const excludedGitHubLabels = slo.appliesTo.excludedGitHubLabels;
+  const hasNoExLabels = await isValidExcludedLabels(
+    issueLabels,
+    excludedGitHubLabels
+  );
+  if (!hasNoExLabels) {
+    console.info(
```

Comment:
meta note: I think I saw @JustinBeckwith recommend we set all our console.logs and console.infos to use the logger from gcf-utils in the GH Automation Chat today. Not sure necessarily where that applies, but just FYI.

Reply:
I was gonna try switching the logging and see if the app is still functional